crates/holochain/src/conductor/conductor/state_dump_helpers.rs

```diff
@@ -0,0 +1,52 @@
+use super::Conductor;
+use crate::conductor::api::error::{ConductorApiError, ConductorApiResult};
+use chrono::{DateTime, Utc};
+use hdk::prelude::CellId;
+use holochain_conductor_api::{AgentInfoDump, P2pAgentsDump};
+use kitsune2_api::AgentInfoSigned;
+use std::sync::Arc;
+
+pub async fn peer_store_dump(
+    conductor: &Conductor,
+    cell_id: &CellId,
+) -> ConductorApiResult<P2pAgentsDump> {
+    let dna_hash = cell_id.dna_hash().clone();
+    let peer_store = conductor
+        .holochain_p2p
+        .peer_store(dna_hash.clone())
+        .await
+        .map_err(|err| ConductorApiError::CellError(err.into()))?;
+    let all_peers = peer_store.get_all().await?;
+    let agent_id = cell_id.agent_pubkey().to_k2_agent();
+    let this_agent_info = peer_store.get(agent_id.clone()).await?;
+    let agent_pub_key = cell_id.agent_pubkey().clone();
+    let space_id = dna_hash.to_k2_space();
+    Ok(P2pAgentsDump {
+        peers: all_peers.into_iter().map(agent_info_dump).collect(),
+        this_agent_info: this_agent_info.map(agent_info_dump),
+        this_agent: Some((agent_pub_key, agent_id)),
+        this_dna: Some((dna_hash, space_id)),
+    })
+}
+
+fn agent_info_dump(peer: Arc<AgentInfoSigned>) -> AgentInfoDump {
+    let created_at = DateTime::from_timestamp_micros(peer.created_at.as_micros())
```

Comment:
The rendering function looked atrocious https://github.com/holochain/holochain/blob/develop/crates/holochain/src/conductor/p2p_agent_store.rs#L301-L340

This gives the same result. Well, a little clearer because hashes are printed as base64 strings instead of byte arrays.